.changes/v0.12.0.md

```diff
@@ -0,0 +1,77 @@
+  Callers relying on a `Container`s default entrypoint will need to be updated
+  to opt-in with `UseEntrypoint`.
+- api: `Container.terminal` now returns a `Container` by @aluzzardi in https://github.com/dagger/dagger/pull/7586 \
+  Callers of `terminal` will need be updated to handle the updated type.
+- api: Return absolute path on export instead of boolean by @helderco in https://github.com/dagger/dagger/pull/7500 \
+  Callers expecting a boolean return will need to be updated to instead
+  handle a string path.
+- api: Don't fallback to the default command on `Container.stdout` and `Container.stderr` by @helderco in https://github.com/dagger/dagger/pull/7857 \
+  Callers of `stdout` and `stderr` without a previous `withExec` will need to
+  insert an explicit empty `withExec`.
+- api: Removed deprecated `sshAuthSocket` and `sshKnownHosts` args from `GitRef.tree` by @jedevc in https://github.com/dagger/dagger/pull/6934 \
+  Callers should instead attach these arguments onto the top-level `git` call.
+- api: Removed `id` parameters for `container`, `directory` and `socket` by @jedevc in https://github.com/dagger/dagger/pull/6934 \
+  Users of these parameters should instead use the standalone
+  `loadContainerFromID`, `loadDirectoryFromID` and `loadSocketFromID`
+  respectively.
+- api: Removed `checkVersionCompatibility` field (versioning checks are now automatically performed on all connections) by @jedevc in https://github.com/dagger/dagger/pull/7751 \
+  Versioning checks are now all automatically performed on all connections, and
+  never need to be manually performed through the API.
+
+#### Upgrade Instructions
+
+Thanks to the new compatibility mode feature, these breaking changes should not
+impact any existing Modules immediately. `dagger call` should still work on
+v0.12.0 without any changes to your Module code (any compat issues for modules
+are likely a bug, please report!)
+
+After upgrading to Engine v0.12.0, you can upgrade your Module to use the
+latest v0.12.0 APIs by running `dagger develop`. That will update the
+`engineVersion` field of your module's `dagger.json` configuration file to
+`v0.12.0` and enable the new APIs.
+
+After that, if your code is impacted by any of the breaking changes, you will
+see errors when running `dagger call`. Once the errors are fixed, `dagger call`
+will work again and your module can be updated in Daggerverse if desired.
+
+More detailed instructions on addressing the individual API incompatible
+changes can be found in the PR descriptions linked above.
+
+### Added
+
+- api: `terminal` can be inserted into the middle of `Container` and `Directory` pipelines to pop an interactive shell by @aluzzardi in https://github.com/dagger/dagger/pull/7586
+- api: Introduced module versioning compatibility by @jedevc in https://github.com/dagger/dagger/pull/7759
+- cli: Add `-q` flag and `DAGGER_QUIET=1` to restore previous verbosity default by @vito in https://github.com/dagger/dagger/pull/7822
```

Comment:
Was this default verbosity changed in v0.11.9? I thought it was in this release.

Reply:
Good point - it changed after v0.11.8 https://github.com/dagger/dagger/issues/7801 (v0.11.9 is right after v0.11.8)

Reply:
I've added another release note here to mention https://github.com/dagger/dagger/pull/7671, where this change comes from.